From 303217f5c498c5006ae30f709bd1e32c71ff0f17 Mon Sep 17 00:00:00 2001
From: Manish Narani <manish.narani@xilinx.com>
Date: Fri, 10 Jan 2020 02:17:29 -0700
Subject: [PATCH 0795/1852] firmware: xilinx: Add DLL reset support

commit 4b6f30d571f8afa069083e69d7ff55a7a4c49f28 from
https://github.com/Xilinx/linux-xlnx.git

Add DLL reset support in ZynqMP firmware driver for SD DLL reset.

Signed-off-by: Manish Narani <manish.narani@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp.c     | 4 ++++
 include/linux/firmware/xlnx-zynqmp.h | 9 ++++++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index 19dc6498e9b2..f7a2fc8e2deb 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -519,6 +519,8 @@ static inline int versal_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_RPU_BOOT_ADDR_CONFIG:
 	case IOCTL_TCM_COMB_CONFIG:
 	case IOCTL_SET_TAPDELAY_BYPASS:
+	case IOCTL_SD_DLL_RESET:
+	case IOCTL_SET_SD_TAPDELAY:
 	case IOCTL_WRITE_GGS:
 	case IOCTL_READ_GGS:
 	case IOCTL_WRITE_PGGS:
@@ -552,6 +554,8 @@ static inline int zynqmp_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_TCM_COMB_CONFIG:
 	case IOCTL_SET_TAPDELAY_BYPASS:
 	case IOCTL_SET_SGMII_MODE:
+	case IOCTL_SD_DLL_RESET:
+	case IOCTL_SET_SD_TAPDELAY:
 	case IOCTL_WRITE_GGS:
 	case IOCTL_READ_GGS:
 	case IOCTL_WRITE_PGGS:
diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index 7d5583d2c27c..c0f7c270dc24 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -158,7 +158,8 @@ enum pm_ioctl_id {
 	IOCTL_TCM_COMB_CONFIG,
 	IOCTL_SET_TAPDELAY_BYPASS,
 	IOCTL_SET_SGMII_MODE,
-	IOCTL_SET_SD_TAPDELAY = 7,
+	IOCTL_SD_DLL_RESET,
+	IOCTL_SET_SD_TAPDELAY,
 	IOCTL_SET_PLL_FRAC_MODE,
 	IOCTL_GET_PLL_FRAC_MODE,
 	IOCTL_SET_PLL_FRAC_DATA,
@@ -352,6 +353,12 @@ enum tap_delay_type {
 	PM_TAPDELAY_OUTPUT,
 };
 
+enum dll_reset_type {
+	PM_DLL_RESET_ASSERT,
+	PM_DLL_RESET_RELEASE,
+	PM_DLL_RESET_PULSE,
+};
+
 enum pm_pinctrl_config_param {
 	PM_PINCTRL_CONFIG_SLEW_RATE,
 	PM_PINCTRL_CONFIG_BIAS_STATUS,
-- 
2.31.1

